assets/src/stories-editor/components/editor-carousel/indicator.js

```diff
@@ -29,27 +57,109 @@ const Indicator = ( { pages, currentPage, onClick } ) => {
 	/* translators: %s: Page number */
 	const toolTip = ( pageNumber ) => sprintf( __( 'Go to page %s', 'amp' ), pageNumber );
 
+	const [ isDragging, setIsDragging ] = useState( false );
+	const { movePageToPosition, initializePageOrder } = useDispatch( 'amp/story' );
+	const { getBlockOrder, getBlockIndex } = useSelect( ( select ) => {
+		return select( 'core/block-editor' );
+	} );
+
 	return (
 		<ul className="amp-story-editor-carousel-item-list">
 			{ pages.map( ( page, index ) => {
 				const className = page.clientId === currentPage ? 'amp-story-editor-carousel-item amp-story-editor-carousel-item--active' : 'amp-story-editor-carousel-item';
+				const blockElementId = `amp-story-editor-carousel-item-${ page.clientId }`;
+				const transferData = {
+					type: 'indicator',
+					srcIndex: getBlockIndex( page.clientId ),
+					srcClientId: page.clientId,
+				};
+
+				const getInsertIndex = ( position ) => {
+					const dstIndex = getBlockIndex( page.clientId );
+					if ( page.clientId !== undefined ) {
+						return position.x === 'left' ? dstIndex : dstIndex + 1;
+					}
+
+					return undefined;
+				};
+
+				const onDrop = ( event, position ) => {
+					const { srcClientId, srcIndex, type } = parseDropEvent( event );
+
+					const isIndicatorDropType = ( dropType ) => dropType === 'indicator';
+					const isSameBlock = ( src, dst ) => src === dst;
```

Comment:
Why is this a function? Why not calculate the actual value and put it in there?

```suggestion
					const isSameBlock = srcClientId === page.clientId;
```

(similarly for `isIndicatorDropType` above)